app/workers/retire_subject_worker.rb

```diff
@@ -1,30 +1,28 @@
 class RetireSubjectWorker
   include Sidekiq::Worker
+  attr_reader :workflow
 
   sidekiq_options queue: :high
 
   def perform(workflow_id, subject_ids, reason=nil)
-    workflow = Workflow.find(workflow_id)
-    retired_subjects = 0
+    begin
+      @workflow = Workflow.find(workflow_id)
+    rescue ActiveRecord::RecordNotFound
+      return nil
+    end
 
-    Workflow.transaction do
-      subject_ids.each do |subject_id|
-        begin
-          if workflow.retire_subject(subject_id, reason)
-            retired_subjects += 1
-          end
-        rescue ActiveRecord::RecordInvalid
-        end
-      end
+    Array.wrap(subject_ids).each do |subject_id|
+      count = subject_workflow_status(subject_id)
+      RetirementWorker.perform_async(count.id, reason)
```

Comment:
This worker has a default of "nil". I think this line will then pass in an explicit `nil` for `reason`, which will override the default value of `classification_count` defined in `RetirementWorker`?

Reply:
this was the previous behaviour so i kept it (see the diff for the details). The operation validates a reason if it's passed in but does allow a nil one too.